src/controllers/message.js

```diff
@@ -29,7 +29,18 @@ const getAllMessages = async (req, res) => {
     const { userId } = req.user
     const messages = await Message.find({
         $or: [{ postedByUser: userId }, { receivedByUser: userId }], //We find all those messages which our userId sent or received. .
-    }).sort({ createdAt: 1 })
+    })
+        .populate({
+            path: 'postedByUser',
+            select: 'username -_id',
+            model: 'User',
+        })
+        .populate({
+            path: 'receivedByUser',
+            select: 'username -_id',
+            model: 'User',
+        })
+        .sort({ createdAt: 1 })
 
     const groupedMessages = messages.reduce((acc, curr) => {
         //here we are checking whether the userId is recipient or sender, otherUser would be the other one!
```

Comment:
Other user will need to be changed. It should be curr.postedByUser.id.tostring() or curr.receivedByUser.toString() I think. Otherwise you end up with the return below where the key is messed up.

![image](https://user-images.githubusercontent.com/70294155/235555389-7d976aa9-e7c0-4a05-a27e-a3da34c09c2b.png)

Reply:
@jpremo could you please let me know where you get this error? I tested it on Postman, it works fine without modifying the otherUser by changing it to `curr.postedByUser.id.toString() or curr.receivedByUser.id.toString()`. 

Then I tested it with `curr.postedByUser.id.toString() or curr.receivedByUser.id.toString()` it returned the same as the original code.